tests/components/mqtt/test_common.py

```diff
@@ -119,6 +119,30 @@ async def help_setup_component(
     return mqtt_mock
 
 
+def help_custom_config(
+    domain: str,
+    default_config: ConfigType,
+    option_update: ConfigType | tuple[ConfigType,],
```

Comment:
Let's try to make this clearer, how about this:
```suggestion
    mqtt_base_config: ConfigType,
    mqtt_entity_domain: str,
    mqtt_entity_configs: Iterable[ConfigType,],
```

if `entity` is not good, maybe `mqtt_component_domain` + `mqtt_component_configs`?